src/Promotion/Action/UnitsFixedPricePromotionActionCommand.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusPromotionExtensionsPlugin\Promotion\Action;
+
+use Setono\SyliusPromotionExtensionsPlugin\Distributor\MostExpensiveFirstDistributorInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Core\Promotion\Action\UnitDiscountPromotionActionCommand;
+use Sylius\Component\Core\Promotion\Filter\FilterInterface;
+use Sylius\Component\Promotion\Model\PromotionInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+final class UnitsFixedPricePromotionActionCommand extends UnitDiscountPromotionActionCommand
+{
+    public const TYPE = 'units_fixed_price';
+
+    /** @var FilterInterface */
+    private $priceRangeFilter;
+
+    /** @var FilterInterface */
+    private $taxonFilter;
+
+    /** @var FilterInterface */
+    private $productFilter;
+
+    /** @var MostExpensiveFirstDistributorInterface */
+    private $distributor;
+
+    public function __construct(
+        FactoryInterface $adjustmentFactory,
+        FilterInterface $priceRangeFilter,
+        FilterInterface $taxonFilter,
+        FilterInterface $productFilter,
+        MostExpensiveFirstDistributorInterface $distributor
+    ) {
+        parent::__construct($adjustmentFactory);
+
+        $this->priceRangeFilter = $priceRangeFilter;
+        $this->taxonFilter = $taxonFilter;
+        $this->productFilter = $productFilter;
+        $this->distributor = $distributor;
+    }
+
+    public function execute(PromotionSubjectInterface $subject, array $configuration, PromotionInterface $promotion): bool
```

Comment:
Does this method have to return a bool? Is that bool used anywhere?

Reply:
That is the interface requirement